client/my-sites/themes/controller.js

```diff
@@ -49,27 +45,43 @@ function getProps( context ) {
 	};
 
 	return {
-		title,
 		tier,
 		search: context.query.s,
 		trackScrollPage: boundTrackScrollPage,
 		runClientAnalytics
 	};
-}
+};
 
-function makeElement( ThemesComponent, Head, store, props ) {
-	return(
-		<ReduxProvider store={ store }>
-			<Head title={ props.title } tier={ props.tier || 'all' }>
-				<ThemesComponent { ...omit( props, [ 'title', 'runClientAnalytics' ] ) } />
+const LoggedInHead = ( { children, context: { params: { tier, site_id: siteID } } } ) => (
+	<Head
+		title={ buildTitle(
+			i18n.translate( 'Themes', { textOnly: true } ),
+			{ siteID }
+		) }
+		tier={ tier || 'all' }>
+		{ children }
+	</Head>
+);
+
+// This is generic -- nothing themes-specific in here!
+function makeElement( Component, getProps, Head, sideEffects = function() {} ) {
+	// How do we handle the dispatch?
+	return ( context, next ) => {
```

Comment:
Yeah, returning middleware here is what I had in mind for an interface (see https://github.com/Automattic/wp-calypso/pull/3425/files#diff-85f90b61d9c36634c5464a3180470726R83). I'm with renaming, to, uh, `makeElementMiddleware` (?) tho. 

This way, `Head` (as a parameter to that function) can be a component that just accepts `context` and generically extracts any props it needs, as opposed to hard-wiring it to `tier` etc which doesn't really scale to any other sections (and not even to theme sheets). Also, we're able to automatically extract the `store` from `context` (and pass it on to `ReduxProvider`).